src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -571,17 +570,29 @@
                     peStream: new MemoryStream(EmittedAssemblyData.ToArray()),
                     options: PdbToXmlOptions.ResolveTokens |
                              PdbToXmlOptions.ThrowOnError |
-                             PdbToXmlOptions.ExcludeDocuments |
                              PdbToXmlOptions.ExcludeCustomDebugInformation |
-                             PdbToXmlOptions.ExcludeScopes,
-                    methodName: sequencePoints);
+                             PdbToXmlOptions.ExcludeScopes |
+                             PdbToXmlOptions.IncludeTokens);
 
                 if (actualPdbXml.StartsWith("<error>"))
                 {
-                    throw new Exception($"Failed to extract PDB information for method '{sequencePoints}'. PdbToXmlConverter returned:{Environment.NewLine}{actualPdbXml}");
+                    throw new Exception($"Failed to extract PDB information. PdbToXmlConverter returned:{Environment.NewLine}{actualPdbXml}");
                 }
 
-                markers = ILValidation.GetSequencePointMarkers(actualPdbXml, source);
+                var methodDef = (Cci.IMethodDefinition)methodData.Method.GetCciAdapter();
+                var methodToken = MetadataTokens.GetToken(_testData.MetadataWriter.GetMethodDefinitionOrReferenceHandle(methodDef));
+                var xmlDocument = XElement.Parse(actualPdbXml);
+                var xmlMethod = ILValidation.GetMethodElement(xmlDocument, methodToken);
+
+                // method may not have any debug info and thus no sequence points
+                if (xmlMethod != null)
+                {
+                    var documentMap = ILValidation.GetDocumentIdToPathMap(xmlDocument);
+
+                    markers = sequencePointsSource ?
+                        ILValidation.GetSequencePointMarkers(xmlMethod, id => _compilation.SyntaxTrees.FirstOrDefault(tree => tree.FilePath == documentMap[id])?.ToString()) :
+                        ILValidation.GetSequencePointMarkers(xmlMethod);
+                }
```

Comment:
In the alternative (null `xmlMethod`) should we assert that the call was not asking for sequence points or sequence points source?

Reply:
`xmlMethod` being null is entirely valid scenario - it just means there are no sequence points. E.g. you can use `VerifyMethodBody` on a method that does not have sequence points (e.g. compiler generated one).

Reply:
BTW, if the method does not exist `var methodData = _testData.GetMethodData(qualifiedMethodName);` will throw.